src/equations/shallow_water_two_layer_2d.jl

```diff
@@ -0,0 +1,850 @@
+that contains the gradient of the bottom topography [`TwoLayerShallowWaterEquations2D`](@ref). This
+is a slightly modified version to account for the additional source term compared to standard SWE.
+
+Further details are available in the paper:
+- Niklas Wintermeyer, Andrew R. Winters, Gregor J. Gassner and David A. Kopriva (2017)
+  An entropy stable nodal discontinuous Galerkin method for the two dimensional
+  shallow water equations on unstructured curvilinear meshes with discontinuous bathymetry
+  [DOI: 10.1016/j.jcp.2017.03.036](https://doi.org/10.1016/j.jcp.2017.03.036)
+"""
+@inline function flux_nonconservative_wintermeyer_etal(u_ll, u_rr, 
+                                                       orientation::Integer,
+                                                       equations::TwoLayerShallowWaterEquations2D)
+  # Pull the necessary left and right state information
+  h1_ll, h2_ll = waterheight(u_ll, equations)
+  h1_rr, h2_rr = waterheight(u_rr, equations)
+  b_rr = u_rr[7]
+
+  z = zero(eltype(u_ll))
+
+  # Bottom gradient nonconservative term: (0, g*h1*(b + h2)_x, g*h1*(b + h2)_y ,
+  #                                        0, g*h2*(b + r*h1)_x, g*h2*(b + r*h1)_y, 0)
+  if orientation == 1
+    f = SVector(z,
+    equations.gravity * h1_ll * (b_rr + h2_rr),
+    z,z,
+    equations.gravity * h2_ll * (b_rr + equations.rho1/equations.rho2 * h1_rr),
+    z,z)
+  else # orientation == 2
+    f = SVector(z, z,
+    equations.gravity * h1_ll * (b_rr + h2_rr),
+    z,z,
+    equations.gravity * h2_ll * (b_rr + equations.rho1/equations.rho2 * h1_rr),
+    z)
+  end
+
+  return f
+end
+
+@inline function flux_nonconservative_wintermeyer_etal(u_ll, u_rr,
+                                                       normal_direction_ll::AbstractVector,
+                                                       normal_direction_average::AbstractVector,
+                                                       equations::TwoLayerShallowWaterEquations2D)
+  # Pull the necessary left and right state information
+  h1_ll, h2_ll = waterheight(u_ll, equations)
+  h1_rr, h2_rr = waterheight(u_rr, equations)
+  b_rr = u_rr[7]
+
+  # Note this routine only uses the `normal_direction_average` and the average of the
+  # bottom topography to get a quadratic split form DG gradient on curved elements
+  return SVector(zero(eltype(u_ll)),
+                normal_direction_average[1] * equations.gravity * h1_ll * (b_rr +  h2_rr),
+                normal_direction_average[2] * equations.gravity * h1_ll * (b_rr +  h2_rr),
+                zero(eltype(u_ll)),
+                normal_direction_average[1] * equations.gravity * h2_ll * (b_rr + 
+                                                            equations.rho1/equations.rho2 * h1_rr),
+                normal_direction_average[2] * equations.gravity * h2_ll * (b_rr +
+                                                            equations.rho1/equations.rho2 * h1_rr),
```

Comment:
Similar to comment above. We could precompute and store `r = equations.rho1/equations.rho2` if it is more convenient. That or we can always use `equations.rho1/equations.rho2` everywhere.